overlord/hookstate/ctlcmd/set.go

```diff
@@ -79,6 +79,11 @@ func (s *setCommand) Execute(args []string) error {
 
 		transaction.Set(s.context().SnapName(), key, value)
 	}
+	context.Lock()
+	if context.IsSnapContext() {
+		transaction.Commit()
```

Comment:
Yes, when the snapctl call comes in, we should test if IsEphemeral right before returning and call its Done method right there if that's the case. The `*Context` won't be saved anywhere, so it'll be garbage collected shortly after.